base/tools/src/main/java/com/netscape/cmstools/nss/NSSCertRequestCLI.java

```diff
@@ -180,24 +217,70 @@ public void execute(CommandLine cmd) throws Exception {
         X509Key subjectKey = CryptoUtil.createX509Key(keyPair.getPublic());
         extensions = generator.createExtensions(subjectKey);
 
-        PKCS10 pkcs10 = nssdb.createPKCS10Request(
-                keyPair,
-                subject,
-                false,
-                hash,
-                extensions);
-
-        String format = cmd.getOptionValue("format");
         byte[] bytes;
 
-        if (format == null || "PEM".equalsIgnoreCase(format)) {
-            bytes = CertUtil.toPEM(pkcs10).getBytes();
+        if ("pkcs10".equalsIgnoreCase(requestType)) {
+
+            PKCS10 pkcs10 = nssdb.createPKCS10Request(
+                    keyPair,
+                    subject,
+                    false,
+                    hash,
+                    extensions);
+
+            if (requestFormat == null || "PEM".equalsIgnoreCase(requestFormat)) {
+                bytes = CertUtil.toPEM(pkcs10).getBytes();
+
+            } else if ("DER".equalsIgnoreCase(requestFormat)) {
+                bytes = pkcs10.toByteArray();
+
+            } else {
+                throw new Exception("Unsupported format: " + requestFormat);
+            }
+
+        } else if ("crmf".equalsIgnoreCase(requestType)) {
+
+            String transportCertPath = cmd.getOptionValue("transport");
+            X509Certificate transportCert = nssdb.addPEMCertificate(transportCertPath, null);
+
+            Name subjectName = CryptoUtil.createName(subject, subjectEncoding);
```

Comment:
IIUC the transport certificate is imported in NSSDB. If a new request is performed the certificate overwrite the request? If the certificate has been already imported, could be available the option for the nick?

Reply:
Good point, let me take a look...

Reply:
The existing tools (`CRMFPopClient` and `pki client-cert-request`) are actually doing the same thing and to my understanding it doesn't cause any problem (the cert already in NSS database will be intact), but I went ahead and change `pki nss-cert-request` to take the cert nickname only so the user is responsible to import it first. Please see the updated PR. Thanks!

Reply:
I did a small test of importing multiple times a certificate with the same nick. If the certificate and the nick match no problem but if the nick is associated with a different certificate then the command fails. Therefore, the command was working if the user has not imported the certificate or if it is imported the same. If the user would requests to different CAs/KRAs having different transport certificate should fail. I think the change to use the nick here is better. Thanks

Reply:
Good to know, thanks!